src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -810,6 +815,50 @@ private SqlCommand BuildGetChangesCommand(SqlConnection connection, SqlTransacti
             return new SqlCommand(getChangesQuery, connection, transaction);
         }
 
+        /// <summary>
+        /// Returns the number of changes(rows) on the user's table that are actively locked by other leases.
+        /// </summary>
+        /// <param name="connection">The connection to add to the SqlCommand</param>
+        /// <param name="transaction">The transaction to add to the SqlCommand</param>
+        /// <returns>The number of rows locked by leases</returns>
+        private async Task<int> GetLeaseLockedRowCount(SqlConnection connection, SqlTransaction transaction)
+        {
+            string leasesTableJoinCondition = string.Join(" AND ", this._primaryKeyColumns.Select(col => $"c.{col.name.AsBracketQuotedString()} = l.{col.name.AsBracketQuotedString()}"));
+            int leaseLockedRowsCount = 0;
+            long getLockedRowCountDurationMs = 0L;
+            // Get the count of changes from CHANGETABLE that meet the following criteria:
+            // * Not Null LeaseExpirationTime AND
+            // * LeaseExpirationTime > Current Time
+            string getLeaseLockedrowCountQuery = $@"
+                {AppLockStatements}
+
+                DECLARE @last_sync_version bigint;
+                SELECT @last_sync_version = LastSyncVersion
+                FROM {GlobalStateTableName}
+                WHERE UserFunctionID = '{this._userFunctionId}' AND UserTableID = {this._userTableId};
+
+                SELECT COUNT(*)
+                FROM CHANGETABLE(CHANGES {this._userTable.BracketQuotedFullName}, @last_sync_version) AS c
+                LEFT OUTER JOIN {this._leasesTableName} AS l ON {leasesTableJoinCondition}
+                WHERE l.{LeasesTableLeaseExpirationTimeColumnName} IS NOT NULL AND l.{LeasesTableLeaseExpirationTimeColumnName} > SYSDATETIME()";
+            try
+            {
+                using (var getLeaseLockedRowCountCommand = new SqlCommand(getLeaseLockedrowCountQuery, connection, transaction))
+                {
+                    var commandSw = Stopwatch.StartNew();
+                    leaseLockedRowsCount = (int)await getLeaseLockedRowCountCommand.ExecuteScalarAsyncWithLogging(this._logger, CancellationToken.None);
+                    getLockedRowCountDurationMs = commandSw.ElapsedMilliseconds;
+                }
+            }
+            catch (Exception ex)
+            {
+                this._logger.LogError($"Failed to query count of lease locked changes for table '{this._userTable.FullName}' due to exception: {ex.GetType()}. Exception message: {ex.Message}");
+                TelemetryInstance.TrackException(TelemetryErrorName.GetLeaseLockedRowCount, ex, null, new Dictionary<TelemetryMeasureName, double>() { { TelemetryMeasureName.GetLockedRowCountDurationMs, getLockedRowCountDurationMs } });
+                throw;
```

Comment:
Since this is just for debugging we shouldn't throw here - I'd suggest just returning -1 in that case so it's clear something went wrong but the rest of the trigger continues to run as expected.

Reply:
Make sure to call that out in the doc comment too in case we use this anywhere else for some reason.

Reply:
return -1 instead of throwing the exception wouldn't help us right since we only log >0. I can remove the throw statement and continue the execution.

Reply:
Oh - and to respond to this. That's fine since we'll still log if the rows returned is >= 0. And since we log the error it'll be visible in the logs regardless.